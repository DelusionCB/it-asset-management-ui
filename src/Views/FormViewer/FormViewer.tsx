import React, {useEffect, useState} from 'react';
import {useLocation, useParams} from 'react-router-dom';
import {getItemData} from '../../Api/Utils/fetchUtils';
import DisplayFields from '../../Components/FormFields/DisplayFields';
import {Spinner} from 'reactstrap';

function FormViewer (): JSX.Element {
    const params = useLocation();
    const directParams = useParams();
    const [item, setItem] = useState<any>()
    const [loading, isLoading] = useState(true)

    const setData = (): void => {
        if (!params.state) {
            void getItemData({itemId: directParams.id, apiPath: directParams.path}).then((res) => {
                setItem(res)
                isLoading(false)
            })
        } else {
            void getItemData({itemId: params.state.id, apiPath: params.state.api}).then((res) => {
                setItem(res)
                isLoading(false)
            })
        }
    }

    useEffect(() => {
        setData()
    }, [])

    if (loading) {
        return (
            <Spinner animation="border" role="status">
                Ladataan...
            </Spinner>
        )
    } else {
        return (
            <div>
                <DisplayFields
                    item={item}
                    type={params.state ? params.state.api : directParams.path}
                />
            </div>
        )
    }
}

export default FormViewer;
