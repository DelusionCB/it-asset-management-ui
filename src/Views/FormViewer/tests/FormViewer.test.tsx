import React from 'react';
import * as TestRenderer from 'react-test-renderer';
import FormViewer from '../FormViewer';
import {BrowserRouter} from 'react-router-dom'
import '@testing-library/jest-dom'

describe('<FormViewer />', () => {
    let renderer: TestRenderer.ReactTestRenderer;

    // Render the component before tests
    beforeAll(() => {
        renderer = TestRenderer.create(
            <BrowserRouter>
                <FormViewer />
            </BrowserRouter>,
        );
    });
    describe('Snapshot', () => {
        it('renders correctly', () => {
            expect(renderer.toJSON()).toMatchSnapshot();
        })
    })
})
