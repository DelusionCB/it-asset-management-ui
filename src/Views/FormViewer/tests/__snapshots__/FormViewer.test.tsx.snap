// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormViewer /> Snapshot renders correctly 1`] = `
<div
  animation="border"
  className="spinner-border"
  role="status"
>
  <span
    className="visually-hidden"
  >
    Ladataan...
  </span>
</div>
`;
